[nat-ong555] iP

nat-ong555 opened the pull request:
# **Pluto**

Pluto frees your mind of having to remember every single thing to do. It's
- text-based
- easy to use

All you need to do is:
1. download it from here
2. double-click it
3. add your tasks
4. let Pluto manage it for you!

And it is **FREE!**

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Setting reminders (coming soon)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

nat-ong555 committed:
Renamed chatbot + added greeting

nat-ong555 committed:
Echo user commands

nat-ong555 committed:
changed to OOP style

nat-ong555 committed:
added TaskList class to maintain the task list

nat-ong555 committed:
added ability to change status of a task

nat-ong555 committed:
added the 3 subclasses of Task

nat-ong555 committed:
handle errors using exception class

nat-ong555 committed:
added feature that can delete tasks

nat-ong555 committed:
changed input.txt and EXPECTED.txt

nat-ong555 committed:
updated runtest.sh

nat-ong555 committed:
Add save tasks and load tasks function

nat-ong555 committed:
Add file that loads tasks

nat-ong555 committed:
Change Deadline and Event tasks to display date format

nat-ong555 committed:
Merge branch 'branch-Level-7'

nat-ong555 committed:
Merge branch 'branch-Level-8'

nat-ong555 committed:
Divide classes into package

nat-ong555 committed:
Merge branch 'A-Packages'

nat-ong555 committed:
Merge branch 'add-gradle-support'

nat-ong555 committed:
Add JUnit tests to test Storage and ToDo classes

nat-ong555 committed:
Create JAR file for chatbot

nat-ong555 committed:
Add JavaDoc comments to methods and constructors

nat-ong555 committed:
Change coding style to comply with Java Coding Standard

nat-ong555 committed:
Update chatbot to process find command, and refactor code to more OOP style

nat-ong555 committed:
Merge branch 'branch-A-CodingStandard'

nat-ong555 committed:
Merge branch 'branch-A-JavaDoc'

nat-ong555 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/pluto/Pluto.java

Kaidama97 commented on src/main/java/pluto/Deadline.java:
```diff
@@ -0,0 +1,45 @@
+     * @return a String to be stored in tasks file
+     */
+    @Override
+    public String toFileFormat() {
```
That's a good function name. descriptive and follows the naming convention 👍

Kaidama97 commented on src/main/java/pluto/Pluto.java:
```diff
@@ -0,0 +1,53 @@
+            try {
+                input = scanner.nextLine();
+                if (input.equals("bye")) {
+                    message.showGoodbyeMessage();
```
Good job on naming. Used method name used verbs and written in camelCase 💯

Kaidama97 commented on src/main/java/pluto/Parser.java:
```diff
@@ -0,0 +1,87 @@
+package pluto;
+
+public class Parser {
+    private static final String COMMAND_TODO = "todo";
+    private static final String COMMAND_DEADLINE = "deadline";
+    private static final String COMMAND_EVENT = "event";
+    private static final String COMMAND_MARK = "mark";
+    private static final String COMMAND_UNMARK = "unmark";
+    private static final String COMMAND_LIST = "list";
+    private static final String COMMAND_DELETE = "delete";
+    private static final String COMMAND_FIND = "find";
```
Good use of style for the static final variables.  May consider using enumeration for your variables?

Kaidama97 commented on src/test/java/pluto/StorageTest.java:
```diff
@@ -0,0 +1,43 @@
+    Path tempDir;
+
+    @Test
+    public void loadTasks_nonExistentFile_newEmptyList() {
```
Good naming of the test method 👍

Kaidama97 reviewed (commented):
Overall naming convension looks good and no issues 🎆

m0destly commented on src/main/java/pluto/Message.java:
```diff
@@ -0,0 +1,72 @@
+     * Displays the welcome message when bot is started up
+     */
+    public void showWelcomeMessage() {
+        String greeting = "Hi! I am pluto.Pluto\n" + "What can I do for you today? \n\n";
```
Seems like "pluto." was appended before Pluto during packaging, might want to remove?

m0destly commented on src/main/java/pluto/Task.java:
```diff
@@ -0,0 +1,89 @@
+    public static Task fromFileFormat(String line) {
+        String[] parts = line.split(" \\| ");
+        boolean isDone = parts[1].equals("1");
+        switch (parts[0]) {
+            case "T":
+                return new ToDo(parts[2], isDone);
+            case "D":
+                return new Deadline(parts[2], parts[3], isDone);
+            case "E":
+                return new Event(parts[2], parts[3], parts[4], isDone);
+            default:
+                return null;
```
Use of switch statement keeps code clean, looks good

davidchongg replied:
There should not be indentation for case clauses.

m0destly commented on src/test/java/pluto/ToDoTest.java:
```diff
@@ -0,0 +1,20 @@
+
+public class ToDoTest {
+
+    @Test
+    public void todoTask_markedAsDone_success() {
+        ToDo task = new ToDo("Return book");
+        task.markAsDone();
+        assertEquals(" [X] Return book", task.taskStatusMessage());
+    }
+
+    @Test
+    public void newToDoTask_markedAsUndone_success() {
+        ToDo task = new ToDo("Submit assignment");
+        assertEquals(" [ ] Submit assignment", task.taskStatusMessage());
+    }
```
Looks well tested

m0destly reviewed (commented):
Looks great, good job

nat-ong555 committed:
Add checkstyle files to check style

davidchongg commented on src/main/java/pluto/Parser.java:
```diff
@@ -0,0 +1,87 @@
+        String command = parts[0].trim().toLowerCase();
+
+        switch (command) {
+        case COMMAND_LIST:
```
Good job following coding standard for switch case statements where case statements are not indented.

davidchongg commented on src/main/java/pluto/Deadline.java:
```diff
@@ -0,0 +1,45 @@
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+public class Deadline extends Task {
```
Perhaps can leave an empty line after import statements.

davidchongg commented on src/main/java/pluto/Storage.java:
```diff
@@ -0,0 +1,67 @@
+package pluto;
+
+import java.io.File;
```
Good job with the ordering of import statements.

davidchongg commented on src/main/java/pluto/Parser.java:
```diff
@@ -0,0 +1,87 @@
+            break;
+        case COMMAND_DEADLINE:
+            if (parts.length < 2 || !parts[1].contains(" /by ")) {
+                throw new PlutoException("The deadline format is incorrect. Please use: deadline <task> /by <yyyy-mm-dd>");
```
Perhaps could use line wrapping for lines longer than 120 characters.

nat-ong555 committed:
Edit files to follow CheckStyle

nat-ong555 committed:
Merge branch 'branch-A-CheckStyle'

nat-ong555 committed:
Add GUI to the chatbot

nat-ong555 committed:
Merge branch 'branch-Level-10'

nat-ong555 committed:
Add assertions for task index validation and parser inputs

The chatbot's Parser class processes user commands and manipulates the
task list. There are also potential issues where invalid task indices
could be accessed, leading to runtime errors

Let's:
- Add assertions to Parser to verify task indices before accessing them
- Add assertions to TaskList to verify task indices and whether the
tasks are null

nat-ong555 committed:
Refactor code quality following Java Coding Standards

- Add JavaDocs to GUI classes
- Make sure variables and methods follow naming conventions
- Comments describe WHAT, WHY and not HOW

nat-ong555 committed:
Merge pull request #1 from nat-ong555/branch-A-Assertions

Add assertions for task index validation and parser inputs

nat-ong555 committed:
Merge pull request #2 from nat-ong555/branch-A-CodeQuality

Refactor code quality following Java Coding Standards

nat-ong555 committed:
Add command schedule for BCD-Extension

This command lets users view the list of tasks for a specific input date
- Use "schedule <yyyy-mm-dd>" format

nat-ong555 committed:
Merge branch 'branch-B-ViewSchedules'

nat-ong555 committed:
Update README.md

nat-ong555 committed:
Update README.md

nat-ong555 committed:
Update README.md

nat-ong555 committed:
Update README.md

nat-ong555 committed:
Update README.md

nat-ong555 committed:
Update README.md

nat-ong555 committed:
Update README.md

nat-ong555 committed:
Update README.md

nat-ong555 committed:
Add files via upload

nat-ong555 committed:
Update README.md

nat-ong555 committed:
Update README.md

nat-ong555 committed:
Update README.md

nat-ong555 committed:
Update README.md

nat-ong555 committed:
Update README.md

nat-ong555 committed:
Update README.md

nat-ong555 committed:
Update README.md

nat-ong555 committed:
Update README.md

nat-ong555 committed:
Update README.md

nat-ong555 committed:
Refactor code to correct weak SLAP

nat-ong555 committed:
Merge branch 'master' of https://github.com/nat-ong555/ip

nat-ong555 committed:
Update build.gradle

nat-ong555 committed:
Update build.gradle

nat-ong555 committed:
Update README.md

nat-ong555 committed:
Update README.md

nat-ong555 committed:
Update code to change chatbot's styling

nat-ong555 committed:
Update Ui.png

nat-ong555 committed:
Update README.md
